src/db.c

```diff
@@ -190,14 +228,30 @@ robj *lookupKeyWriteOrReply(client *c, robj *key, robj *reply) {
  * The program is aborted if the key already exists. */
 void dbAdd(redisDb *db, robj *key, robj *val) {
     sds copy = sdsdup(key->ptr);
-    dictEntry *de = dictAddRaw(db->dict, copy, NULL);
+    int slot = getKeySlot(key->ptr);
+    dict *d = db->dict[slot];
+    dictEntry *de = dictAddRaw(d, copy, NULL);
     serverAssertWithInfo(NULL, key, de != NULL);
-    dictSetVal(db->dict, de, val);
+    dictSetVal(d, de, val);
+    db->key_count++;
+    if (server.cluster_enabled) {
+        cumulativeKeyCountAdd(db, slot, 1);
+    }
```

Comment:
```suggestion
    cumulativeKeyCountAdd(db, slot, 1);
```
This check is already in cumulativeKeyCountAdd

Reply:
True, it's not necessary, I was hoping to avoid extra function call when cluster mode is disabled, but the cost must be negligible.